patches/server/0961-Call-EntityDeathEvent-for-EnderDragon-kill.patch

```diff
@@ -0,0 +1,24 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Alfie-Joe Smith <alfie@alfiesmith.net>
+Date: Fri, 10 Feb 2023 06:30:11 +0000
+Subject: [PATCH] Call EntityDeathEvent for EnderDragon#kill
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/boss/enderdragon/EnderDragon.java b/src/main/java/net/minecraft/world/entity/boss/enderdragon/EnderDragon.java
+index c7caaebfb4b9f28cbe700d88fdcf232a500e8ca7..3c3a99a87094515b37a441ce3d9a3b6ef9fe3e57 100644
+--- a/src/main/java/net/minecraft/world/entity/boss/enderdragon/EnderDragon.java
++++ b/src/main/java/net/minecraft/world/entity/boss/enderdragon/EnderDragon.java
+@@ -616,6 +616,13 @@ public class EnderDragon extends Mob implements Enemy {
+ 
+     @Override
+     public void kill() {
++        // Paper start
++        org.bukkit.event.entity.EntityDeathEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callEntityDeathEvent(this, drops);  // Paper - call event
```

Comment:
I'm not sure we should pass in the `drops` field here. It looks like the ender dragon should drop nothing here, so passing in a list that might have something in it, might not is misleading.

Also, the callEntityDeathEvent method inside CraftEventFactory has some logic for playing death sounds. I think you need to set the `silentDeath` field on LivingEntity to true before calling the event for EnderDragon so it doesn't play the death sound by default (which would be a behavior change).

Reply:
Will fix this in a few hours. I passed in the drops as a convention (the same is done for armor stand, but I suppose that has drops) - surely it is always empty for the dragon?

Will change the silentDeath field also.

Reply:
Yeah, as far as I can tell it is always going to be empty there, but we should just pass an empty list regardless, to its clear its always empty. If you pass the drops list, it might look like its supposed to have drops.